pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -115,8 +117,27 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 		isIPv6 = true
 	}
 
-	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
-	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
+	sIface, srcIsLocal := c.ifaceStore.GetInterfaceByIP(srcIP)
+	dIface, dstIsLocal := c.ifaceStore.GetInterfaceByIP(dstIP)
+
+	isFlexibleIPAMSrc, isFlexibleIPAMDst, ctZone, err := parseFlexibleIPAMStatus(pktIn, c.nodeConfig, srcIP, srcIsLocal, dstIP, dstIsLocal)
+	if err != nil {
+		return err
+	}
+	matches := pktIn.GetMatches()
+	if (isFlexibleIPAMSrc || isFlexibleIPAMDst) && dstIsLocal {
```

Comment:
More questions:

1. Is it better to use: `isFlexibleIPAMSrc && dstIsLocal || isFlexibleIPAMDst`?
2. When `isFlexibleIPAMSrc || isFlexibleIPAMDst` is false, it should not be possible that `dstIsLocal` is true, but the packet is from uplink/gateway?

Reply:
1. Sure. FIxed.
2. A local NodePort/ServiceWithoutAntreaProxy access with regular src/dst Pod, which means `isFlexibleIPAMSrc || isFlexibleIPAMDst` is false and `dstIsLocal` is true, and I'm not sure if there is SNAT when the packet is DNATed by Node's iptables. Thus I prefer to restrict this new logic for FlexibleIPAM cases only to avoid changing existing logic.

Reply:
Ok.